pkg/source/adapter/adapter.go

```diff
@@ -177,3 +198,22 @@ func (a *Adapter) Handle(ctx context.Context, msg *sarama.ConsumerMessage) (bool
 func (a *Adapter) SetRateLimits(r rate.Limit, b int) {
 	a.rateLimiter = rate.NewLimiter(r, b)
 }
+
+func (a *Adapter) HandleServiceMessage(ctx context.Context, message ctrl.ServiceMessage) {
+	// In this first PR, there is only the RA sending messages to control plane,
+	// there is no message the control plane should send to the RA
+	a.logger.Info("Received unexpected control message")
+	message.Ack()
+}
+
+func (a *Adapter) Setup(sess sarama.ConsumerGroupSession) {
+	if err := a.controlServer.SendAndWaitForAck(kafkasourcecontrol.NotifySetupClaimsOpCode, kafkasourcecontrol.Claims(sess.Claims())); err != nil {
+		a.logger.Warnf("Cannot send the claims update: %v", err)
```

Comment:
these codecov warnings make me wonder, do we have a way to mock/unittest the control protocol as a consumer?

Reply:
I recently developed some mocks and other testing utilities https://github.com/knative-sandbox/control-protocol/tree/main/pkg/test, but I still don't use these here. You want me to tackle it now or in a followup?

Reply:
Aren't you planning to start using more of that? e.g. for changing/updating topics?

I am fine in handling it via a follow-up.